runtime/v2/services/reflection.go

```diff
@@ -0,0 +1,33 @@
+package services
+
+import (
+	"context"
+
+	"github.com/cosmos/gogoproto/proto"
+	"google.golang.org/protobuf/types/descriptorpb"
+
+	reflectionv1 "cosmossdk.io/api/cosmos/reflection/v1"
+)
+
+// ReflectionService implements the cosmos.reflection.v1 service.
+type ReflectionService struct {
+	reflectionv1.UnimplementedReflectionServiceServer
+	files *descriptorpb.FileDescriptorSet
+}
+
+func NewReflectionService() (*ReflectionService, error) {
+	fds, err := proto.MergedGlobalFileDescriptors()
+	if err != nil {
+		return nil, err
+	}
+
+	return &ReflectionService{files: fds}, nil
+}
+
+func (r ReflectionService) FileDescriptors(_ context.Context, _ *reflectionv1.FileDescriptorsRequest) (*reflectionv1.FileDescriptorsResponse, error) {
+	return &reflectionv1.FileDescriptorsResponse{
+		Files: r.files.File,
+	}, nil
+}
+
+var _ reflectionv1.ReflectionServiceServer = &ReflectionService{}
```

Comment:
taking note for later, but I believe we should move this out of runtime. There isn't a need for this to live in runtime. It would be the underlying server that would set this up